### PR TITLE
To support swagger spec 2.0 the following fields have to be added:

### PR DESCRIPTION
**Pull request by gaalandr**
To support swagger spec 2.0 the following fields have to be added to SerializableParameter and AbstractSerializableParameter

See spec 2.0 params: https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject

missing fields:
``` java
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor17.
	 */
    public Number maximum;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor17
	 */
    public Boolean exclusiveMaximum;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor21
	 */
    public Number minimum;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor21
	 */
    public Boolean exclusiveMinimum;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor26
	 */
    public Integer maxLength;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor29
	 */
    public Integer minLength;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor33
	 */
    public String pattern;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor42
	 */
    public Integer maxItems;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor45
	 */
    public Integer minItems;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor49
	 */
    public Boolean uniqueItems;
	/**
	 * See http://json-schema.org/latest/json-schema-validation.html#anchor14
	 */
    public Number multipleOf;
```

Please verify and merge into master.

### CODE REVIEW

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import java.util.List;
 
 @JsonPropertyOrder({"name", "in", "description", "required", "type", "items", "collectionFormat", "default",
-        "maximum", "exclusiveMaximum", "minimum", "exclusiveMinimum", "maxItems", "minItems"})
+        "maximum", "exclusiveMaximum", "minimum", "exclusiveMinimum", "maxItems", "minItems", "maxLength","minLength","pattern","uniqueItems","multipleOf"})
```
Comment (iushankin) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: order of properties should be same as they are defined in the spec

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java [hunk 2]
```diff
@@ -244,7 +265,55 @@ public Object getExample() {
...
+
+	public Number getMultipleOf() {
+		return multipleOf;
+	}
+
+	public void setMultipleOf(Number multipleOf) {
+		this.multipleOf = multipleOf;
+	}
+
+	public Boolean getExclusiveMaximum() {
```
Comment (iushankin) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: duplicated method - `public Boolean isExclusiveMaximum()`

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java [hunk 3]
```diff
@@ -244,7 +265,55 @@ public Object getExample() {
...
+		return exclusiveMaximum;
+	}
+
+	public Boolean getExclusiveMinimum() {
```
Comment (iushankin) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: duplicated method - `public void setExclusiveMaximum(Boolean exclusiveMaximum)`

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/SerializableParameter.java [hunk 4]
```diff
@@ -24,4 +24,49 @@
...
+	void setMinimum(Double minimum);
+
+	Integer getMaxItems();
+
+	void setMaxItems(Integer maxItems);
+
+	Integer getMinItems();
+
+	void setMinItems(Integer minItems);
+
```
Comment (iushankin) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/SerializableParameter.java]: please add new line to end of file

File: modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import io.swagger.models.properties.Property;
+import io.swagger.models.properties.PropertyBuilder;
+import io.swagger.models.properties.PropertyBuilder.PropertyId;
+
+public class SerializableParameterTest {
+	
+	@Test
+	public void testParameters() throws NoSuchMethodException, SecurityException {
+		
+		String[] requredMethods = {
```
Comment (iushankin) [inline modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java]: Let's wrap this array with `@org.testng.annotations.DataProvider` so we could check every property independently.
As for now if we have more then one skipped properties the test will detect the first one only

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java [hunk 6]
```diff
@@ -244,7 +265,55 @@ public Object getExample() {
         }
         return example;
     }
-    public void setExample(String example) {
+    public Integer getMaxLength() {
```
Comment (iushankin) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: add `@Override` annotation if a method is implementing an interface

**Comment by iushankin** [comment]
Could you please follow the [Swagger Coding Style](https://github.com/swagger-api/swagger-core/wiki/Coding-Standards): tab characters are not used for indentation.

File: modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		
+		String[] methodPrefix = { "get", "set" };
+		
+		for (String methodName : requredMethods) {
+			methodName = capitalize(methodName);
+			for (String mp : methodPrefix) {
+				String methodNameWithPrefix = mp + methodName;
+				boolean found = false;
+				for (Method method : methods) {
+					if (method.getName().equals(methodNameWithPrefix)) {
```
Comment (iushankin) [inline modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java]: Generally speaking this check gives positive results for non-bean methods. (i.e. String setSomeValue(String arg1, String arg2, ...)). Is it possible use the `com.fasterxml.jackson.databind.BeanDescription` class here as we do it in the [ModelResolver](https://github.com/swagger-api/swagger-core/blob/v1.5.4/modules/swagger-core/src/main/java/io/swagger/jackson/ModelResolver.java#L249)?

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java (hunk 1 above)
Comment (gaalandr) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: done

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java (hunk 2 above)
Comment (gaalandr) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: done

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java (hunk 3 above)
Comment (gaalandr) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: I think this is not duplicated

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/SerializableParameter.java (hunk 4 above)
Comment (gaalandr) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/SerializableParameter.java]: done

File: modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java (hunk 5 above)
Comment (gaalandr) [inline modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java]: done

File: modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java (hunk 6 above)
Comment (gaalandr) [inline modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java]: done

File: modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java (hunk 7 above)
Comment (gaalandr) [inline modules/swagger-models/src/test/java/io/swagger/models/parameters/SerializableParameterTest.java]: java.beans.PropertyDescriptor().getWriteMethod(),
java.beans.PropertyDescriptor().getReadMethod

should handle the problem, and give negative results for non methods. Do you agree?

**Comment by gaalandr** [comment]
Hi Iushankin,

I have updated the requests. Please confirm.

**Comment by iushankin** [comment]
Hi @gaalandr, thank you.
Your changes are good, but could you remove tab characters for indentation.
Please look at the [Swagger Coding Style](https://github.com/swagger-api/swagger-core/wiki/Coding-Standards) for details.

**Comment by gaalandr** [comment]
I have already imported the coding styles, but I will check and fix.
Thanks!

**Comment by gaalandr** [comment]
Hi,

can you please merge, I have fixed the indentation problems. (the linked ecplise code style does not work, It does not fix the identation issues.)

**Comment by iushankin** [comment]
@gaalandr thanks

**Pull request merged**